site/_posts/2021-03-16-securely-implement-oauth-vuejs.md

```diff
@@ -1174,32 +1221,310 @@ methods: {
   //
 </form>
 ```
 
-> FUN FACT: Using `.prevent` stops the page from reloading whenever the Submit button is clicked. 
+Also, be sure and update your `App.vue` to account for the user-data.  Here is a snippet of what your `App.vue` should look like from the improvements in this last section.
+
+```vue
+import Greet from './Greeting';
+import Login from "./Login";
+import Update from "./Update";
+
+export default {
+  name: 'app',
+  components: {
+    Greet,
+    Login,
+    Update
+  },
+//...
+```
+
+> ***FUN FACT***: Using `.prevent` stops the page from reloading whenever the Submit button is clicked. 
+
+To summarize, here is a snapshot of the full code for `Update.vue` now:
+```vue
+<template>
+  <form @submit.prevent="update">
+  <textarea
+    v-model="userData"
+    placeholder="Update FusionAuth user data."
+  ></textarea>
+    <button type="submit" class="button">Submit</button>
+  </form>
+</template>
+<script>
+export default {
+  name: "Update",
+  data() {
+    return {
+      userData: "",
+    };
+  },
+  methods: {
+    update: function () {
+      fetch(`http://localhost:9000/set-user-data`, {
+        credentials: "include",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          userData: this.userData,
+        }),
+      }).catch((err) => {
+        console.log(err);
+      });
+      this.userData = ''
+    },
+  },
+}
+</script>
+<style>
+textarea {
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+}
+
+button {
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 5px;
+}
+</style>
+```
 
 Here is how our application looks now:
 
 {% include _image.liquid src="/assets/img/blogs/oauth-vuejs/signed-in-with-text-area.png" alt="The application when logged in with the text area for updating the data." class="img-fluid" figure=false %}
 
 Go to your Vue app and type some text in the `textarea` and click the Submit button. If you log in to the FusionAuth dashboard, you can now see the text you added is stored in FusionAuth. It is under the **User data** tab in your user account details.
 
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/custom-user-data.png" alt="custom user data" class="img-fluid top-cropped" figure=false %}
+
+## One final condition 
+
+If you have made it this far...Great!  You have a fully working example of a VueJS application using FusionAuth.  
+
+The remainder of this article will cover one final condition and improve our existing code.  
+
+### Authorization vs authentication 
+
+What happens if we create a user in FusionAuth but do not assign them to an application?  You may notice our ExpressJS server crashes.
+
+Why is that?  The short answer is we need to refactor our code (namely our axios code).  The longer answer is we need to understand [Authentication vs Authorization](https://fusionauth.io/docs/v1/tech/core-concepts/authentication-authorization/).  
```

Comment:
Do we need two pictures of the admin UI? One with a user registered and one without? We can re-use the one near the text "Next, add our current user to the new application. Select Users on the dashboard, select Manage and go to the Registration tab. Then click Add Registration, and add yourself to the application you just created." for the former. 

But I think it'd be illuminating to have a UI screen shot for the latter case.